lib/pdk/util/git.rb

```diff
@@ -55,8 +56,27 @@ def self.remote_repo?(maybe_repo)
         git('ls-remote', '--exit-code', maybe_repo)[:exit_code].zero?
       end
 
+      def self.work_dir_clean?(repo)
+        raise PDK::CLI::ExitWithError, _('Unable to locate git work dir "%{workdir}') % { workdir: repo } unless File.directory?(repo)
+        raise PDK::CLI::ExitWithError, _('Unable to locate git dir "%{gitdir}') % { gitdir: repo } unless File.directory?(File.join(repo, '.git'))
+
+        git('--work-tree', repo, '--git-dir', File.join(repo, '.git'), 'status', '--untracked-files=no', '--porcelain', repo)[:stdout].empty?
+      end
+
+      def self.current_branch(repo)
+        env = { 'GIT_DIR' => repo }
```

Comment:
For consistency with other git invocations, and debugging purposes (we only print specific environment variables in the debug output), can you switch this over to using the `--git-dir` parameter instead of passing the dir through the environment?

Reply:
The `bare_repo?` method also passes `GIT_DIR` through an env variable. Should that be changed too?

Reply:
I believe the issue with the `bare_repo?` method is that `git rev-parse` has a very understanding of the `--git-dir` option.

(Git CLI inconsistency)--

Reply:
Ah, and since this method also calls `rev-parse`, you are probably stuck using the ENV var.